acra-censor/common/logging_logic_test.go

```diff
@@ -195,79 +195,58 @@ func TestQueryCapture(t *testing.T) {
-			if err != nil {
-				t.Fatal(err)
-			}
-		}
-		expected := "{\"raw_query\":\"SELECT Student_ID FROM STUDENT\",\"_blacklisted_by_web_config\":false}\n" +
-			"{\"raw_query\":\"SELECT * FROM STUDENT\",\"_blacklisted_by_web_config\":false}\n" +
-			"{\"raw_query\":\"SELECT * FROM X\",\"_blacklisted_by_web_config\":false}\n" +
-			"{\"raw_query\":\"SELECT * FROM Y\",\"_blacklisted_by_web_config\":false}\n"
-
-		time.Sleep(DefaultSerializationTimeout + extraWaitTime)
-		result, err := ioutil.ReadFile(tmpFile.Name())
-		if err != nil {
-			t.Fatal(err)
-		}
-		if !strings.EqualFold(strings.ToUpper(string(result)), strings.ToUpper(expected)) {
-			t.Fatal("Expected: " + expected + "\nGot: " + string(result))
-		}
-		testQuery := "SELECT * FROM Z;"
-		_, err = writer.RedactAndCheckQuery(testQuery)
-		if err != nil {
-			t.Fatal(err)
-		}
-		expected = "{\"raw_query\":\"SELECT Student_ID FROM STUDENT\",\"_blacklisted_by_web_config\":false}\n" +
-			"{\"raw_query\":\"SELECT * FROM STUDENT\",\"_blacklisted_by_web_config\":false}\n" +
-			"{\"raw_query\":\"SELECT * FROM X\",\"_blacklisted_by_web_config\":false}\n" +
-			"{\"raw_query\":\"SELECT * FROM Y\",\"_blacklisted_by_web_config\":false}\n" +
-			"{\"raw_query\":\"SELECT * FROM Z\",\"_blacklisted_by_web_config\":false}\n"
-
-		time.Sleep(DefaultSerializationTimeout + extraWaitTime)
-		result, err = ioutil.ReadFile(tmpFile.Name())
-		if err != nil {
-			t.Fatal(err)
-		}
-		if !strings.EqualFold(strings.ToUpper(string(result)), strings.ToUpper(expected)) {
-			t.Fatal("Expected: " + expected + "\nGot: " + string(result))
-		}
-
-		//Check that values are hidden while logging
-		testQuery = "select songName from t where personName in ('Ryan', 'Holly') group by songName having count(distinct personName) = 10"
-
-		writer.RedactAndCheckQuery(testQuery)
-
-		//wait until serialization completes
-		time.Sleep(DefaultSerializationTimeout + extraWaitTime)
-
-		result, err = ioutil.ReadFile(tmpFile.Name())
-		if err != nil {
-			t.Fatal(err)
-		}
 
-		expectedPrefix := "{\"raw_query\":\"SELECT Student_ID FROM STUDENT\",\"_blacklisted_by_web_config\":false}\n" +
-			"{\"raw_query\":\"SELECT * FROM STUDENT\",\"_blacklisted_by_web_config\":false}\n" +
-			"{\"raw_query\":\"SELECT * FROM X\",\"_blacklisted_by_web_config\":false}\n" +
-			"{\"raw_query\":\"SELECT * FROM Y\",\"_blacklisted_by_web_config\":false}\n" +
-			"{\"raw_query\":\"SELECT * FROM Z\",\"_blacklisted_by_web_config\":false}\n" +
-			"{\"raw_query\":\"select songName from t where personName in"
+	for _, query := range testQueries {
+		writer.WriteQuery(query)
+	}
+	expected := "{\"raw_query\":\"SELECT Student_ID FROM STUDENT;\",\"_blacklisted_by_web_config\":false}\n" +
+		"{\"raw_query\":\"SELECT * FROM STUDENT;\",\"_blacklisted_by_web_config\":false}\n" +
+		"{\"raw_query\":\"SELECT * FROM X;\",\"_blacklisted_by_web_config\":false}\n" +
+		"{\"raw_query\":\"SELECT * FROM Y;\",\"_blacklisted_by_web_config\":false}\n"
```

Comment:
we can drop it because it repeat previous or change "Y" -> "X" to test duplicated